Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -464,13 +505,24 @@ public CsharpCompiler(ErrorReporter reporter)
 
                 Contract.Assert(arg.CompileName == dtor.CorrespondingFormals[n - 1].CompileName);
                 wGet.WriteLine(
-                  $"return (({dt.CompileName}_{dtor.EnclosingCtors[n - 1].CompileName}{DtT_TypeArgs})d).{IdName(arg)}; ");
+                  $"return (({dt.CompileName}_{dtor.EnclosingCtors[n - 1].CompileName}{DtT_TypeArgs})d).{IdName(arg)};");
               }
             }
           }
         }
       }
 
+      foreach (var member in dt.Members) {
+        if (member.IsGhost || member.IsStatic) continue;
+        if (member is Function fn && !NeedsCustomReceiver(member)) {
```

Comment:
What scenario is the `NeedsCustomReceiver` check for? Is it possible that this is true here?

Reply:
Dafny and C# have slightly different concepts of variance. Some members that are valid in Dafny, aren't in C#, so they can't be exported through the interface, where we declare the variance. Instead, we rewrite them to be static and call the through the abstract class.